server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexStateService.java

```diff
@@ -349,28 +354,33 @@ private void sendVerifyShardBeforeCloseRequest(final IndexShardRoutingTable shar
     /**
      * Step 3 - Move index states from OPEN to CLOSE in cluster state for indices that are ready for closing.
      */
-    static ClusterState closeRoutingTable(final ClusterState currentState, final Map<Index, AcknowledgedResponse> results) {
+    static ClusterState closeRoutingTable(final ClusterState currentState,
+                                          final ClusterBlock closingBlock,
+                                          final Map<Index, AcknowledgedResponse> results) {
         final MetaData.Builder metadata = MetaData.builder(currentState.metaData());
         final ClusterBlocks.Builder blocks = ClusterBlocks.builder().blocks(currentState.blocks());
         final RoutingTable.Builder routingTable = RoutingTable.builder(currentState.routingTable());
 
         final Set<String> closedIndices = new HashSet<>();
         for (Map.Entry<Index, AcknowledgedResponse> result : results.entrySet()) {
             final Index index = result.getKey();
+            final boolean acknowledged = result.getValue().isAcknowledged();
             try {
                 final IndexMetaData indexMetaData = metadata.getSafe(index);
-                if (indexMetaData.getState() != IndexMetaData.State.CLOSE) {
-                    if (result.getValue().isAcknowledged()) {
+                assert currentState.blocks().hasIndexBlock(index.getName(), closingBlock);
```

Comment:
how do we know that the close block has not been removed? An open command could have come in in the mean-time, removing the close block?

Reply:
I changed the logic here to accommodate with keeping only the last index closed block.

> An open command could have come in in the mean-time, removing the close block?

Opening indices only removes blocks and changes state of indices that are not OPEN so I think we're good and block should not be removed by an open command.